kickstarter/ios-oss
# Fix Content Hugging and Compression in PledgeDisclaimerView.iconImageView

# 📲 What

Fixes alignment issues for the `iconImageView` in `PledgeDisclaimerView` by setting appropriate content hugging and compression resistance priorities.

# 🤔 Why

The `iconImageView` displayed inconsistent alignment on the checkout screen due to:

Variations in the "Kickstarter is not a store" text content.
Differences in the checkout screen components rendered.
Changes in the os language, which altered text length based on translations.
This fix ensures consistent alignment of the `iconImageView` across all scenarios.

# 🛠 How

Updated constraints for the `iconImageView` in `setupConstraints`:
```swift
private func setupConstraints() {
   self.iconImageView.setContentHuggingPriority(.required, for: .horizontal)
   self.iconImageView.setContentCompressionResistancePriority(.required, for: .horizontal)
}
```
Ensures that the `iconImageView` maintains its alignment regardless of the text length, screen components, or system language.

# 👀 See

- [Jira](https://kickstarter.atlassian.net/browse/MBL-1915)
- See snapshots included in this PR

# ✅ Acceptance criteria

- [ ] Steps to test:
   1. Navigate to the checkout screen.
   2. Observe `iconImageView` alignment with various text content lengths and components.
   3. Test alignment in different system languages.
- [ ]  Issues to watch:
   - Confirm alignment under edge cases with mixed components.
